packages/terra-compact-interactive-list/src/CompactInteractiveList.jsx

```diff
@@ -99,9 +124,108 @@ const CompactInteractiveList = (props) => {
     width,
     columnMinimumWidth,
     columnMaximumWidth,
+    onCellSelect,
+    onClearSelection,
   } = props;
 
   const theme = useContext(ThemeContext);
+  const listRef = React.useRef();
+
+  // using ref so that keyboard navigation and focusing on cells won't trigger component re-render.
+  const focusedCell = useRef({ row: 0, cell: 0 });
+
+  const setFocusOnCell = ({ row, cell }) => {
+    // add 1 to the row number to accomodate for hidden header
+    const focusedCellElement = listRef.current.children[row + 1].children[cell];
+    const interactiveChildren = getFocusableElements(focusedCellElement);
+    if (interactiveChildren?.length > 0 && interactiveChildren[0]) {
+      // currently a cell can have only one interact-able element in it, which gets auto-focused.
+      interactiveChildren[0].focus();
+    } else {
+      // cell gets focus if there is no interact-able elements in it.
+      focusedCellElement.focus();
+    }
+  };
+
+  const setFocusedRowCol = ({ row, cell }) => {
+    focusedCell.current = { row, cell };
+    setFocusOnCell({ row, cell });
+  };
+
+  const handleOnCellSelect = useCallback(({ row, cell }) => {
+    focusedCell.current = { row, cell };
+    // check for interactive elements in that cell
+    // add 1 to the row number to accomodate for hidden header
+    const elementToFocus = listRef.current.children[row + 1].children[cell];
+    const interactiveChildren = getFocusableElements(elementToFocus);
+    // if no interactive elements
+    if (onCellSelect && interactiveChildren?.length === 0) {
+      onCellSelect(rows[row].id, columns[cell].id);
+    }
```

Comment:
Why would the presence of interactive elements prevent us from triggering the callback for cell selection?  What happens if the user clicks on the cell but not the interactive element?

Reply:
The intention is for the interactive element to be the only interactive thing in the cell.